pkg/client/kubernetes/scaling_test.go

```diff
@@ -132,6 +133,41 @@ var _ = Describe("scale", func() {
 			Expect(ScaleStatefulSetAndWaitUntilScaled(context.TODO(), runtimeClient, key, 2)).To(Succeed(), "scale done")
 		})
 	})
+
+	Describe("#WaitUntilNoPodRunningForDeployment", func() {
+		It("should wait until there are no running pods for the deployment", func() {
+			selector := &metav1.LabelSelector{
+				MatchLabels: map[string]string{"foo": "bar"},
+			}
+
+			runtimeClient.EXPECT().Get(gomock.Any(), key, gomock.AssignableToTypeOf(&appsv1.Deployment{})).DoAndReturn(
```

Comment:
Please do not use mock client for new tests, ref https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-unit-tests